### PR TITLE
Upgrade Spock to 1.3

### PR DESCRIPTION
**Pull request by paplorinc**
See: http://spockframework.org/spock/docs/1.3/release_notes.html#_what_s_new_in_this_release

> What’s New In This release
> * Add implicit assertions for CodeArgument constraints (https://github.com/spockframework/spock/pull/956)
> * Add power assertion output to asserts with explicit message (https://github.com/spockframework/spock/pull/928)
> * Add support for mixed named and positional arguments in mocks (https://github.com/spockframework/spock/pull/919)
> * Add NamedParam support for groovy-2.5 with backport to 2.4 (https://github.com/spockframework/spock/pull/921)
> * Add special rendering for Set comparisons (https://github.com/spockframework/spock/pull/925)
> * Add identity hash code to type hints in comparison failures if they are identical
> * Fix erroneous regex where an optional colon was defined instead of a non-capturing group (https://github.com/spockframework/spock/pull/931)
> * Improve CodeArgumentConstraint by supporting assertions (https://github.com/spockframework/spock/pull/918)
> * Improve IDE type inference in MockingApi (https://github.com/spockframework/spock/pull/920)
> * Improve reporting of TooFewInvocationsError (https://github.com/spockframework/spock/pull/912)
> * Improve render class loader for classes in comparison failures (https://github.com/spockframework/spock/pull/932)
> * Improve record class literal values to display FQCN in comparison failures (https://github.com/spockframework/spock/pull/935)
> * Improve filter Java 9+ reflection stack frames
> * Improve show stacktrace of throwables in comparison failure result
> * Improve use canonical class name in comparison failure results if present
> * Improve render otherwise irrelevant expressions if they get a type hint in comparison failure (https://github.com/spockframework/spock/pull/936)
> * Fix do not convert implicit "this" expression like when calling the constructor of a non-static inner class (https://github.com/spockframework/spock/pull/930)
> * Fix class expression recording when there are comments with dots in the same line (https://github.com/spockframework/spock/pull/937)

### CODE REVIEW

File: subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/DefaultVersionedComponentChooserTest.groovy [hunk 1]
```diff
@@ -232,6 +233,7 @@ class DefaultVersionedComponentChooserTest extends Specification {
 
     }
 
+    @Ignore(/* Fixme @melix */)
```
Comment (paplorinc) [inline subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/DefaultVersionedComponentChooserTest.groovy]: @melix, need your help here

Comment (paplorinc) [reply subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/DefaultVersionedComponentChooserTest.groovy]: Fixed, the problem was the `with` clause, for some reason its param was `null`.
Luckily it has a `this` as well, used that instead and extracted to a separate methods for clarity.

**Comment by paplorinc** [comment]
Reverted test dependencies.

**Comment by paplorinc** [comment]
Build passed, and checked it in Idea (2019.1.3 only, since 2019.2 EAP has trouble resolving Spock 1.3) that the mock navigations work after the inline (cc: @marcphilipp)

File: subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/DefaultVersionedComponentChooserTest.groovy [hunk 2]
```diff
@@ -252,30 +252,26 @@ class DefaultVersionedComponentChooserTest extends Specification {
...
         }
         1 * selectedComponentResult.doesNotMatchConsumerAttributes({
-            it.id == c.id &&
-                it.matchingDescription.find { it.requestedAttribute == Attribute.of('color', String) }
-                    .with { match ->
-                    assert match.requestedValue.get() == 'red'
-                    assert match.found.get() == 'green'
-                    match
-                }
+            it.id == c.id
```
Comment (wolfs) [inline subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/DefaultVersionedComponentChooserTest.groovy]: Is this a new Spock feature you are using here?

File: subprojects/build-init/src/main/resources/org/gradle/buildinit/tasks/templates/library-versions.properties [hunk 3]
```diff
@@ -10,5 +10,5 @@ scala-xml=1.2.0
 scala=2.12
 scalatest=3.0.7
 slf4j=1.7.26
-spock=1.2-groovy-2.5
+spock=1.3-groovy-2.5
```
Comment (wolfs) [inline subprojects/build-init/src/main/resources/org/gradle/buildinit/tasks/templates/library-versions.properties]: This file shouldn't be changed manually, but via the `updateInitPluginTemplateFile`. In order to update the Spock version, I guess this line needs to be changed: https://github.com/gradle/gradle/blob/8ab371109b8d75117edcb1fa93a3bf92ee764dad/subprojects/build-init/build-init.gradle.kts#L90

File: subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/DefaultVersionedComponentChooserTest.groovy (hunk 2 above)
Comment (paplorinc) [reply subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/DefaultVersionedComponentChooserTest.groovy]: yes, this behaves now as the assert blocks, evaluating each part separately: http://spockframework.org/spock/docs/1.3/release_notes.html#_code_argument_constraints_are_treated_as_implicit_assertions

If I change them they won't just show `false` but power assertion output is shown, see:
Screen Shot 2019-06-05 at 17 32 38
Screen Shot 2019-06-05 at 17 33 59

File: subprojects/build-init/src/main/resources/org/gradle/buildinit/tasks/templates/library-versions.properties (hunk 3 above)
Comment (paplorinc) [reply subprojects/build-init/src/main/resources/org/gradle/buildinit/tasks/templates/library-versions.properties]: Thanks for the review, Stefan!
I've ran `gw updateInitPluginTemplateVersionFile` for https://github.com/gradle/gradle/pull/9486/files#diff-10559753926ae8cd59f961f3aca772c3R90 and don't see any changes locally. Am I missing something?

**Comment by marcphilipp** [comment]
@paplorinc What's the issue in IDEA?

File: subprojects/build-init/src/main/resources/org/gradle/buildinit/tasks/templates/library-versions.properties (hunk 3 above)
Comment (wolfs) [reply subprojects/build-init/src/main/resources/org/gradle/buildinit/tasks/templates/library-versions.properties]: Sorry, looks like I missed the updated line above: https://github.com/gradle/gradle/pull/9486/files#diff-10559753926ae8cd59f961f3aca772c3R90

**Comment by paplorinc** [comment]
@marcphilipp The inner mocks weren't seen by the IDE before. Thanks to your change, now we could get rid of the customized `Specifiation` and the IDE still sees them correctly.

**Pull request merged**